code/datums/uplink_item.dm

```diff
@@ -12,10 +12,16 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 			var/datum/uplink_item/I = new path
 			if(!I.item)
 				continue
-			if(I.gamemodes.len && SSticker && !(SSticker.mode.type in I.gamemodes))
-				continue
-			if(I.excludefrom.len && SSticker && (SSticker.mode.type in I.excludefrom))
-				continue
+			if(I.include_modes.len)
+				if(!gamemode && SSticker.mode && !(SSticker.mode.type in I.include_modes))
```

Comment:
It might make more sense to first check if gamemode is null or not, and then depending on that check either do the two checks for SSticker or for gamemode as appropriate

Reply:
I'd rather not touch this code since I'm not really sure how it works...

Reply:
If....you don't know how it works really, then you probably shouldn't just be refactoring it willy nilly.

Reply:
I want to learn how it works, though!

Reply:
Then you need to learn, first, before refactoring.

Refactoring without having any idea of how/why something works is just asking for problems, partial+buggy+incomplete refactors, or unintended behavior.

Reply:
But this PR is fairly simple and already works as intended, so why not just give a quick explanation here as to why something small should be changed so I can properly change it and be done with it?

Reply:
If it's so simple, then why is it that you say you don't understand what Crazy is talking about and are unwilling to change it?

Reply:
I am unwilling to change it as I don't understand why or how that is needed.
I get why you would check the game mode, but I don't know what ssticker is about. Sure I could just change what he said to change but why not learn WHY do that?